sklearn/inspection/tests/test_permutation_importance.py

```diff
@@ -150,3 +150,26 @@ def test_permutation_importance_linear_regresssion():
                                      scoring='neg_mean_squared_error')
     assert_allclose(expected_importances, results.importances_mean,
                     rtol=1e-1, atol=1e-6)
+
+
+def test_permutation_importance_equivalence_sequential_paralell():
+    # regression test to make sure that sequential and parallel calls will
+    # output the same results.
+    X, y = make_regression(n_samples=500, n_features=10, random_state=0)
+
+    X = scale(X)
+    y = scale(y)
+
+    lr = LinearRegression().fit(X, y)
+
+    importance_parallel = permutation_importance(
+        lr, X, y, n_repeats=5, random_state=0, n_jobs=2
+    )
+    importance_sequential = permutation_importance(
+        lr, X, y, n_repeats=5, random_state=0, n_jobs=1
+    )
+
+    assert_allclose(
+        importance_parallel['importances'],
+        importance_sequential['importances']
+    )
```

Comment:
Maybe add a check that we did not end up with trivial solutions:

```python
# Check that the problem is structured enough and that the model is complex enough
# to not yield trivial, constant importances:
imp_min = importance_sequential['importances'].min()
imp_max = importance_sequential['importances'].max()
assert imp_max - imp_min > 0.1
```

I have not check the 0.1 threshold but I guess that would be on the right scale for r2 score? Maybe not.

Reply:
I addressed my own comment.